### ISSUE TITLE
Question: What determines the execution order of plugins

### ISSUE THREAD

**Issue by elidupuis** [svg/svgo#589]
Just curious if there's any way to control the execution order of plugins... I ran into a conflict where one plugin was trying target elements by `id` but the `removeAttrs` plugin had already been run—and it was configured to remove the `id` attribute!  Obviously, the order the plugins are executed would make a difference in this case...

**Comment by GreLI** [comment]
Plugins are being run in order they're defined in [config](https://github.com/svg/svgo/blob/master/docs/how-it-works/en.md#1-config). By default it's defined in [.svgo.yml](/svg/svgo/blob/master/.svgo.yml).

**Comment by elidupuis** [comment]
Thanks for clarifying, @GreLI. Turns out that my issue was actually the `cleanupIDs` plugin running before the `removeElementsByAttr`—which I had configured to remove an element by ID but by that time the ID had already been removed.

Easy fix was to explicitly declare the `cleanupIDs` plugin in my config *after* the `removeElementsByAttr` plugin.

I wonder if it's worth giving `removeElementsByAttr` a little more priority in the order of execution... Seems like there could definitely be conflicts with `cleanupAttrs` and `cleanupIDs`.
